Casbin.UnitTest/MultithreadingTests/MultithreadingTests.cs

```diff
@@ -0,0 +1,316 @@
+using System.Collections.Generic;
+using System.Linq;
+using System.Threading;
+using System.Threading.Tasks;
+using Casbin.Model;
+using Casbin.UnitTests.Fixtures;
+using Xunit;
+using static Casbin.UnitTests.Util.TestUtil;
+
+namespace Casbin.UnitTests.MultithreadingTests
+{
+    internal class ReaderWriterPolicyManagerForTest : ReaderWriterPolicyManager
+    {
+        private readonly bool _waitWhenWrite;
+        private readonly bool _waitWhenRead;
+        private readonly List<string> _result;
+
+        public ReaderWriterPolicyManagerForTest(IPolicyStore policyStore, List<string> result, bool waitWhenWrite, bool waitWhenRead) : base(policyStore)
+        {
+            _result = result;
+            _waitWhenWrite = waitWhenWrite;
+            _waitWhenRead = waitWhenRead;
+        }
+
+        public static IPolicyManager Create(List<string> result, bool waitWhenWrite, bool waitWhenRead)
+        {
+            return new ReaderWriterPolicyManagerForTest(DefaultPolicyStore.Create(), result, waitWhenWrite, waitWhenRead);
+        }
+
+        public override Task<bool> RemovePolicyAsync(string section, string policyType, IEnumerable<string> rule)
```

Comment:
It overwrites all content, we may not yet get the error even though the base is broken, can we mock an adapter to test?